launch/launch/actions/execute_local.py

```diff
@@ -597,6 +597,10 @@ def prepare(self, context: LaunchContext):
             # pid is added to the dictionary in the connection_made() method of the protocol.
         }
 
+        if self.__respawn:
+            self.__respawn = evaluate_condition_expression(
+                context, normalize_to_list_of_substitutions(self.__respawn))
```

Comment:
@kosuke55 hmm, we should use `normalize_typed_substitution` and `perform_typed_substitution` (see [here](https://github.com/ros2/launch/blob/b59bb6b88b40eaa82870542d93bdfb1a033d05c3/launch/launch/utilities/type_utils.py#L366)).

Reply:
@hidmic Used them. https://github.com/ros2/launch/pull/569